application/src/main/java/org/togetherjava/tjbot/commands/BotCommandAdapter.java

```diff
@@ -0,0 +1,140 @@
+package org.togetherjava.tjbot.commands;
+
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import net.dv8tion.jda.api.events.interaction.component.ButtonInteractionEvent;
+import net.dv8tion.jda.api.events.interaction.component.SelectMenuInteractionEvent;
+import net.dv8tion.jda.api.interactions.commands.Command;
+import net.dv8tion.jda.api.interactions.commands.build.CommandData;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.togetherjava.tjbot.commands.componentids.ComponentId;
+import org.togetherjava.tjbot.commands.componentids.ComponentIdGenerator;
+import org.togetherjava.tjbot.commands.componentids.Lifespan;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Adapter implementation of a {@link BotCommand}. The minimal setup only requires implementation of
+ * their respective command method. A new command can then be registered by adding it to
+ * {@link Features}.
+ * <p>
+ * Further, {@link #onButtonClick(ButtonInteractionEvent, List)} and
+ * {@link #onSelectionMenu(SelectMenuInteractionEvent, List)} can be overridden if desired. The
+ * default implementation is empty, the adapter will not react to such events.
+ * <p>
+ * <p>
+ * The adapter manages some getters for you, you've to create the {@link CommandData} yourself. See
+ * {@link #BotCommandAdapter(CommandData, CommandVisibility)}} for more info on that. Minimal
+ * modifications can be done on the {@link CommandData} returned by {@link #getData()}.
+ * <p>
+ * <p>
+ * If implementations want to add buttons or selection menus, it is highly advised to use component
+ * IDs generated by {@link #generateComponentId(String...)}, which will automatically create IDs
+ * that are valid per {@link SlashCommand#onSlashCommand(SlashCommandInteractionEvent)}.
+ * <p>
+ * <p>
+ * Some example commands are available in {@link org.togetherjava.tjbot.commands.basic}.
+ * Registration of commands can be done in {@link Features}.
+ */
+public abstract class BotCommandAdapter implements BotCommand {
+    private final String name;
+    private final Command.Type type;
+    private final CommandVisibility visibility;
+    private final CommandData data;
+    private ComponentIdGenerator componentIdGenerator;
+
+    /**
+     * Creates a new adapter with the given data.
+     *
+     * @param data the data for this command
+     * @param visibility the visibility of the command
+     */
+    protected BotCommandAdapter(@NotNull CommandData data, @NotNull CommandVisibility visibility) {
+        this.data = Objects.requireNonNull(data, "The data shouldn't be null");
+        this.visibility = Objects.requireNonNull(visibility, "The visibility shouldn't be null");
```

Comment:
u have annotated it with `@NotNull` already then y use `Object.requireNonNull`?

Reply:
The annotation doesn't do a null check

Reply:
The annotation should not exist, see review comment